shared/actions/fs/index.js

```diff
@@ -315,6 +303,26 @@ function* ignoreFavoriteSaga(action: FsGen.FavoriteIgnorePayload): Saga.SagaGene
   }
 }
 
+function* fileActionPopup(action: FsGen.FileActionPopupPayload): Saga.SagaGenerator<any, any> {
+  const {path, type, targetRect} = action.payload
+  // We may not have the folder loaded yet, but will need metadata to know
+  // folder entry types in the popup. So dispatch an action now to load it.
+  type === 'folder' && (yield Saga.put(FsGen.createFolderListLoad({path})))
+  yield Saga.put(
```

Comment:
same here

Reply:
So this task (`fileActionPopup`) is actually triggered in multiple places. How do I match a path using `putActionIfOnPath`?

Reply:
Oh I guess that could be part of the payload.